platform-sdk/consensus-otter-tests/docs/container-environment.md

```diff
@@ -180,20 +180,43 @@ the module. The image is built using the Eclipse Temurin base image for Java 21:
 
 ```dockerfile
 FROM eclipse-temurin:21
-RUN mkdir /opt/DockerApp && \
-    mkdir /opt/DockerApp/apps && \
-    mkdir /opt/DockerApp/lib
-COPY apps/* /opt/DockerApp/apps/
-COPY lib/* /opt/DockerApp/lib/
-EXPOSE 8080
+
+# Create non-root user and group
+RUN groupadd -r appuser && useradd -r -g appuser appuser
+
+# Create app directories and set ownership
+RUN mkdir -p /opt/DockerApp/apps /opt/DockerApp/lib \
+    && chown -R appuser:appuser /opt/DockerApp
+
+# Copy app artifacts with proper ownership
+COPY --chown=appuser:appuser apps/* /opt/DockerApp/apps/
+COPY --chown=appuser:appuser lib/* /opt/DockerApp/lib/
+
+# Run as non-root user
+USER appuser
+
 CMD ["java", "-jar", "/opt/DockerApp/apps/DockerApp.jar"]
 ```
 
 ### Container Startup Process
 
-When a `ContainerNode` is created, it starts a Docker container running the `DockerApp`. The container is configured to
-expose port 8080 for gRPC communication. The `DockerApp` initializes a gRPC server and listens for incoming connections
-from the `ContainerNode`.
+There are two processes, each with their own API, that the `ContainerNode` interacts with:
+
+1. **DockerApp**: The application running inside the Docker container that starts and stops the second process via a
```

Comment:
Maybe add that this is the App that gets called from the start and runs until the container gets killed